tests/testServer/testServer.cpp

```diff
@@ -1,72 +1,181 @@
+        std::cout << "Stopping server" << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
 
 int main(int argc, char **argv)
 {
 
-    if(argc >= 2 and (std::string(argv[1]) == "-h" or std::string(argv[1]) == "--help"))
+    if (argc >= 2 and (std::string(argv[1]) == "-h" or std::string(argv[1]) == "--help"))
     {
-        std::cout << "A simple gRPC test server implementing RPCs using most of the proto3 language features." << std::endl << std::endl;
+        std::cout << "A simple gRPC test server implementing RPCs using most of the proto3 language features." << std::endl
+                  << std::endl;
         std::cout << "SYNOPSIS:" << std::endl;
-        std::cout << "testServer [OPTIONS] [PORT]" << std::endl << std::endl;
+        std::cout << "testServer [OPTIONS] [PORT]" << std::endl
+                  << std::endl;
         std::cout << "OPTIONS:" << std::endl;
         std::cout << "  -h" << std::endl;
         std::cout << "  --help" << std::endl;
-        std::cout << "     Shows this help" << std::endl << std::endl;
+        std::cout << "     Shows this help" << std::endl
+                  << std::endl;
         std::cout << "PORT:" << std::endl;
         std::cout << "  The TCP port the server should listen to." << std::endl;
         std::cout << "  Default: 50051" << std::endl;
         return 0;
     }
-    std::string addr = "0.0.0.0:50051";
-    if(argc >=2)
+    //std::string serverAddr = "0.0.0.0:50051";
+    std::string secureServerAddr = "localhost:50443";
+    std::string dfltServerAddr = "localhost:50052";
+    std::string insecureServerAddr = "localhost:50051";
+    if (argc >= 2)
+    {
+        //serverAddr = "0.0.0.0:" + std::string(argv[1]);
+        insecureServerAddr = "localhost:" + std::string(argv[1]);
+        std::cout << argv[1] << std::endl;
+    }
+
+    //std::cout << argv[1] << std::endl;
+
+    // Frage: Wo wird der Channel geöffnet? Hier create ich doch nur einen Insecure Server
+    // Wie enable ich TLS für den Server? Ich habe da keinen Parameter gefunden. Genügt es über den TLS Port zu gehen?:
+    // std::shared_ptr< grpc::SslServerCredentials > secureCredentials
+    // builder.AddListeningPort(addr+":465", creds )
+
+    // std::cout << "Starting server listening on " << serverAddr << std::endl;
+    // grpc::ServerBuilder builder;
+    // builder.AddListeningPort(serverAddr, grpc::InsecureServerCredentials());
+
+    if (argv[1] == "-ssl")
+    {
+        std::cout << "Flag is working" << std::endl;
+    }
+    else
     {
-        addr = "0.0.0.0:" + std::string(argv[1]);
+        std::cout << "Flag is not working" << std::endl;
+
+        std::cout << insecureServerAddr << std::endl;
     }
-    std::cout << "Starting server listening on " << addr << std::endl;
+    std::cout << "Starting secure server listening on " << secureServerAddr << std::endl;
+    std::cout << "Starting server for secur channel listening on " << dfltServerAddr << std::endl;
```

Comment:
not clear what difference is between secure server and secur channel ?

Reply:
Added more extensive explanation